substrafl/dependency/schemas.py

```diff
@@ -101,3 +136,126 @@ def copy_dependencies_local_code(self, *, dest_dir: Path) -> List[Path]:
+        dependencies_paths = self.copy_dependencies_local_package(dest_dir=local_dep_dir)
+
+        for dependency in dependencies_paths:
+            if dependency.__str__().endswith(".whl"):
+                self._wheels.append(local_dep_dir.relative_to(dest_dir) / str(dependency))
+            else:
+                wheel_name = manage_dependencies.build_user_dependency_wheel(
+                    Path(dependency),
+                    dest_dir=local_dep_dir,
+                )
+                self._wheels.append(local_dep_dir.relative_to(dest_dir) / wheel_name)
+
+        # Local code management
+        self._local_paths = self.copy_dependencies_local_code(dest_dir=dest_dir)
+
+        # requirement.txt edition
+        if self.compile:
+            # pip-compile the requirements.in into a requirements.txt
+            manage_dependencies.compile_requirements(self.pypi_dependencies + self._wheels, dest_dir=dest_dir)
+        else:
+            manage_dependencies.write_requirements(self.pypi_dependencies + self._wheels, dest_dir=dest_dir)
+
+    def _compute_in_cache_directory(self) -> None:
+        """Compute the dependencies in an cache directory.
+        Use ``_delete_cache_directory`` method to delete it.
+        Use ``_copy_cache_directory`` method to copy it in an other directory.
+        """
+        if self._cache_directory is None:
+            self._cache_directory = Path(tempfile.mkdtemp(dir=str(Path.cwd().resolve()), prefix=TMP_SUBSTRAFL_PREFIX))
+            self._compute(self._cache_directory)
+        else:
+            logger.warning(
+                f"Dependencies already computed in cache at {self._cache_directory}. Run `_delete_cache_directory` to "
+                "delete it."
+            )
+
+    def _copy_cache_directory(self, dest_dir: Path) -> None:
+        """Copy the cache computed dest_dir content to the dest_dir given as argument.
+        The method ``_compute_in_cache_directory`` must be run before calling this method.
+        An error will be raised if not.
+
+        Args:
+            dest_dir (Path): directory where to copy the content of the cache directory.
+
+        Raises:
+            exceptions.DependencyCacheNotFoundError: the method ``_compute_in_cache_directory`` must be
+                run before trying to copy the cache directory.
+        """
+        if self._cache_directory is not None:
+            shutil.copytree(self._cache_directory, dest_dir, dirs_exist_ok=True)
+        else:
+            raise exceptions.DependencyCacheNotFoundError(
+                "No cache directory found for the dependencies. Have you computed the dependencies?"
+            )
+
+    def _delete_cache_directory(self) -> None:
+        """Delete the cache directory if it exists. A warning message is send to the logger
+        if the cache directory does not exist.
+        """
+        if self._cache_directory is not None:
+            shutil.rmtree(self._cache_directory)  # delete directory
```

Comment:
We don't want this function to raise any Exception because it will be called in a `__del__`, so it's probably best to wrap the `shutil.rmtree` in a `try except`